libs/api/domains/directorate-of-labour/src/lib/directorate-of-labour.repository.ts

```diff
@@ -1,58 +1,64 @@
 import { Injectable } from '@nestjs/common'
 import { logger } from '@island.is/logging'
+import {
+  UnionApi,
+  Union,
+  PensionApi,
+  PensionFund,
+} from '@island.is/vmst-client'
 import { ParentalLeavePeriod } from './parentalLeavePeriod.model'
-import { Union } from './union.model'
-import { PensionFund } from './pensionFund.model'
 import { ParentalLeaveEntitlement } from './parentalLeaveEntitlement.model'
 import { ParentalLeavePaymentPlan } from './parentalLeavePaymentPlan.model'
 
-// const accessToken = process.env.DIRECTORATE_OF_LABOUR_ACCESS_TOKEN
-
-// TODO implement this class when the endpoints are ready, this is just mocks atm
+const isRunningInDevelopment = process.env.NODE_ENV === 'development'
 
 @Injectable()
 export class DirectorateOfLabourRepository {
-  //   private client: any
-
-  constructor() {
+  constructor(private unionApi: UnionApi, private pensionApi: PensionApi) {
     logger.debug('Created Directorate of labour repository')
   }
 
-  //   getClient(): any {
-  //     if (!accessToken) {
-  //       throw new Error(
-  //         'Missing environment variables: DIRECTORATE_OF_LABOUR_ACCESS_TOKEN',
-  //       )
-  //     }
+  async getUnions(): Promise<Union[]> {
+    if (isRunningInDevelopment) {
+      return [
+        {
+          id: 'id',
+          name: 'VR',
+        },
+      ]
+    }
 
-  //     if (this.client) {
-  //       return this.client
-  //     }
+    const { unions } = await this.unionApi.unionGetUnions()
 
-  //     return 'new client'
-  //   }
+    if (unions) {
+      return unions
+    }
 
-  async getUnions(): Promise<Union[]> {
-    return [
-      {
-        id: 'id',
-        name: 'VR',
-      },
-    ]
+    throw new Error('Could not fetch unions')
   }
 
   async getPensionFunds(): Promise<PensionFund[]> {
-    return [
-      {
-        id: 'id',
-        name: 'Frjalsi',
-      },
-    ]
+    if (isRunningInDevelopment) {
+      return [
+        {
+          id: 'id',
+          name: 'Frjalsi',
+        },
+      ]
+    }
+
+    const { pensionFunds } = await this.pensionApi.pensionGetPensionFunds()
+
+    if (pensionFunds) {
+      return pensionFunds
+    }
+
+    throw new Error('Could not fetch pension funds')
```

Comment:
Is `pensionFunds` nullable? I'd hope `pensionApi` would throw a better error instead.

Reply:
That would be ideal.